src/test/java/seedu/address/logic/commands/transaction/EditTransactionCommandTest.java

```diff
@@ -0,0 +1,209 @@
+
+        assertCommandSuccess(editTransactionCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_duplicateTransactionUnfilteredList_failure() {
+        Transaction firstTransaction = model.getFilteredTransactionList().get(INDEX_FIRST_TRANSACTION.getZeroBased());
+        EditTransactionDescriptor descriptor = new EditTransactionDescriptorBuilder(firstTransaction).build();
+        EditTransactionCommand editTransactionCommand = new EditTransactionCommand(INDEX_SECOND_TRANSACTION,
+                descriptor);
+
+        assertCommandFailure(editTransactionCommand, model, EditTransactionCommand.MESSAGE_DUPLICATE_TRANSACTION);
+    }
+
+    @Test
+    public void execute_duplicateTransactionFilteredList_failure() {
+        showTransactionAtIndex(model, INDEX_FIRST_TRANSACTION);
+
+        // edit transaction in filtered list into a duplicate in address book
+        Transaction transactionInList = model.getUniCash().getTransactionList()
+                .get(INDEX_SECOND_TRANSACTION.getZeroBased());
+        EditTransactionCommand editTransactionCommand = new EditTransactionCommand(INDEX_FIRST_TRANSACTION,
+                new EditTransactionDescriptorBuilder(transactionInList).build());
+
+        assertCommandFailure(editTransactionCommand, model, EditTransactionCommand.MESSAGE_DUPLICATE_TRANSACTION);
+    }
+
+    @Test
+    public void execute_invalidTransactionIndexUnfilteredList_failure() {
+        Index outOfBoundIndex = Index.fromOneBased(model.getFilteredTransactionList().size() + 1);
+        EditTransactionDescriptor descriptor = new EditTransactionDescriptorBuilder()
+                .withName(VALID_TRANSACTION_NAME_NUS).build();
+        EditTransactionCommand editTransactionCommand = new EditTransactionCommand(outOfBoundIndex, descriptor);
+
+        assertCommandFailure(editTransactionCommand, model,
+                UniCashMessages.MESSAGE_INVALID_TRANSACTION_DISPLAYED_INDEX);
+    }
+
+    /**
+     * Edit filtered list where index is larger than size of filtered list,
+     * but smaller than size of address book
+     */
+    @Test
+    public void execute_invalidTransactionIndexFilteredList_failure() {
+        showTransactionAtIndex(model, INDEX_FIRST_TRANSACTION);
+        Index outOfBoundIndex = INDEX_SECOND_TRANSACTION;
+        // ensures that outOfBoundIndex is still in bounds of address book list
+        assertTrue(outOfBoundIndex.getZeroBased() < model.getUniCash().getTransactionList().size());
+
+        EditTransactionCommand editTransactionCommand = new EditTransactionCommand(outOfBoundIndex,
+                new EditTransactionDescriptorBuilder().withName(VALID_TRANSACTION_NAME_NUS).build());
+
+        assertCommandFailure(editTransactionCommand, model,
+                UniCashMessages.MESSAGE_INVALID_TRANSACTION_DISPLAYED_INDEX);
+    }
+
+    @Test
+    public void equals() {
```

Comment:
Is it possible to add a unit test for `EditPersonDescriptor` to fix the code coverage?